jupyterhub_traefik_proxy/proxy.py

```diff
@@ -75,13 +75,35 @@ def _concurrency_changed(self, change):
         Has no effect if `Proxy.should_start` is False.
         """,
     )
+
     extra_dynamic_config = Dict(
         config=True,
         help="""Extra dynamic configuration for treafik.
 
         Merged with the default dynamic config during startup.
 
-        Always takes effect.
+        Always takes effect unless should_start and enable_setup_dynamic_config are both False.
+        """,
+    )
+
+    enable_setup_dynamic_config = Bool(
+        True,
+        config=True,
+        help="""
+        Whether to initialize the traefik dynamic configuration
+        from JupyterHub configuration.
+
+        Creates the traefik API router and TLS setup, if any.
+        When True, only traefik static configuration must be managed by the external service
+        (configuration of the endpoints and provider).
```

Comment:
Not entirely true. Other traefik services, routers etc. can be configured using any one of traefik's available dynamic configuration providers. e.g. I mainly use [docker labels](https://doc.traefik.io/traefik/providers/docker/) on services that I want traefik to route, including jupyterhub and the traefik API settings I already have configured.

Further, the traefik file provider can be configured to watch a directory, into which other dynamic configuration options can be added. I use a mix of both docker labels and files, as not all options are available in all providers.

So, maybe instead:-
When True, the traefik API entrypoint and router must not already be configured outside of jupyterhub (e.g. in the traefik container's docker labels or another one of traefik's dynamic configuration provider).

Reply:
Thanks, I'll try to clarify that only static config is required, and the api router should not be setup twice.